newrelic/structures_newrelic_workload.go

```diff
@@ -0,0 +1,163 @@
+package newrelic
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/newrelic/newrelic-client-go/pkg/nerdgraph"
+	"github.com/newrelic/newrelic-client-go/pkg/workloads"
+)
+
+func expandWorkload(d *schema.ResourceData) workloads.Workload {
+	workload := workloads.Workload{
+		Name: d.Get("name").(string),
+		ID:   d.Get("workload_id").(int),
+		GUID: d.Get("guid").(string),
+	}
+
+	if e, ok := d.GetOk("entity"); ok {
+		workload.Entities = expandWorkloadEntities(e.(*schema.Set).List())
+	}
+
+	if e, ok := d.GetOk("entity_search_query"); ok {
+		workload.EntitySearchQueries = expandWorkloadEntitySearchQueries(e.(*schema.Set).List())
+	}
+
+	if e, ok := d.GetOk("scope_accounts"); ok {
+		workload.ScopeAccounts = expandWorkloadScopeAccounts(e.([]interface{})[0].(map[string]interface{}))
```

Comment:
This could be nil so we should do a couple checks here
```
  if len(e) > 0 && e[0] != nil {
    workload.ScopeAccounts = expandWorkloadScopeAccounts(e.([]interface{})[0].(map[string]interface{}))
  }
```